#1. State the Hypotheses:
Null Hypothesis:
 There is no association between the type of smart home device purchased and the customer satisfaction level.

Alternative Hypothesis:
There is an association between the type of smart home device purchased and the customer satisfaction level.

In [4]:
#2. Compute the Chi-Square Statistic:
import numpy as np
from scipy.stats import chi2_contingency

# Define the observed frequencies as a 2D array
observed = np.array([
    [50, 70],   # Very Satisfied
    [80, 100],  # Satisfied
    [60, 90],   # Neutral
    [30, 50],   # Unsatisfied
    [20, 50]    # Very Unsatisfied
])

# Perform the Chi-Square test
chi2, p, dof, expected = chi2_contingency(observed)

# Print the results
print(f"Chi-Square Statistic: {chi2}")
print(f"P-value: {p}")
print(f"Degrees of Freedom: {dof}")
print("Expected Frequencies:")
print(expected)


Chi-Square Statistic: 5.638227513227513
P-value: 0.22784371130697179
Degrees of Freedom: 4
Expected Frequencies:
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


In [3]:
from scipy.stats import chi2

# Define the significance level and degrees of freedom
alpha = 0.05
df = 4

# Find the critical value
critical_value = chi2.ppf(1 - alpha, df)

print(f"Critical Value: {critical_value}")


Critical Value: 9.487729036781154


 #4. Make a Decision:
If the Chi-Square statistic is greater than the critical value, reject the null hypothesis.
If the Chi-Square statistic is less than or equal to the critical value, fail to reject the null hypothesis.

Since 6.728 < 9.488, the Chi-Square statistic is less than the critical value.

Fail to reject the null hypothesis.

Conclusion:
There is no significant association between the type of smart home device purchased (Smart Thermostat vs. Smart Light)
and the customer's satisfaction level at the 0.05 significance level. This means that the observed differences in satisfaction levels for the two types of devices are not statistically significant.